### PR TITLE
Remove jaeger-agent from distributions

### PR DESCRIPTION
**Pull request by yurishkuro**
## Which problem is this PR solving?
- Part of #4739

## Description of the changes
- Remove jaeger-agent from binary and Docker distributions
- Move agent's port mappings into `cmd/agent/app/ports.go` to clean up the main ports

## Dependencies
- Need to refactor crossdock to not rely on agent

## How was this change tested?
- CI

## Checklist
- [ ] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [ ] I have signed all commits
- [ ] I have added unit tests for the new functionality
- [ ] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: cmd/all-in-one/all_in_one_test.go [hunk 1]
```diff
@@ -20,24 +21,28 @@ import (
...
-	queryPort  = "16686"
-	agentPort  = "5778"
-	healthPort = "13133"
-	queryAddr  = "http://" + host + ":" + queryPort
-	agentAddr  = "http://" + host + ":" + agentPort
-	healthAddr = "http://" + host + ":" + healthPort + "/status"
+	host         = "0.0.0.0"
+	queryPort    = ports.QueryHTTP
+	samplingPort = ports.CollectorHTTP
+	healthPort   = 13133
```
Comment (yurishkuro) [inline cmd/all-in-one/all_in_one_test.go:34]: TODO: move to pkg/ports

File: cmd/all-in-one/all_in_one_test.go [hunk 2]
```diff
@@ -20,24 +21,25 @@ import (
...
 	getTraceURL            = "/api/traces/"
 	getServicesAPIV3URL    = "/api/v3/services"
-	getSamplingStrategyURL = "/sampling?service=whatever"
+	getSamplingStrategyURL = "/api/sampling?service=whatever"
+)
+
+var (
+	queryAddr    = fmt.Sprintf("http://%s:%d", host, ports.QueryHTTP)
+	samplingAddr = fmt.Sprintf("http://%s:%d", host, ports.CollectorHTTP)
+	healthAddr   = fmt.Sprintf("http://%s:%d/status", host, ports.CollectorV2HealthChecks)
```
Comment (mahadzaryab1) [inline cmd/all-in-one/all_in_one_test.go:42]: just a question for my own understanding: why is this the port here being called `CollectorV2HealthChecks` when it is being used in v1 as well?

**Review by mahadzaryab1 — APPROVED** [review]
looks great! just one question

File: cmd/all-in-one/all_in_one_test.go (hunk 2 above)
Comment (yurishkuro) [reply cmd/all-in-one/all_in_one_test.go:42]: it's not being used with v1:
```
func healthCheckV2(t *testing.T) {
	if os.Getenv("HEALTHCHECK_V2") == "false" {
		t.Skip("Skipping health check for V1 Binary")
	}
```

**Pull request merged**